src/lseq/ident.rs

```diff
@@ -1,78 +1,103 @@
+use crate::vclock::Actor;
+use bitvec::vec::*;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
-use bitvec::vec::*;
 
 const BOUNDARY: u64 = 10;
 
 const INITIAL_BASE: u32 = 3; // start with 2^8
 
-/// A tree identifier uniquely locates a character in an LSeq tree
-/// it represents the path that needs to be taken in order to reach
+/// Common identifier trait for an LSEQ. The type should implement
+/// [`Actor`] and [`Default`] to be usable as an identifier.
+pub trait Ident: Actor + Default {}
```

Comment:
the name `Ident` seems wrong here. It's still an `Actor` just with a sentinel value. On top of that there's an `Identifier` trait in LSEQ already.

Reply:
Hmmm ok, I think I'm starting to understand the needs here better. How about this: we introduce a new type, say, `SentinelValue`. This is similar to `Option` where we have an empty case, and the other case contains a value. The empty case acts as our sentinel value that does nothing but is used for comparisons as a lower bound.

```rust
#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
enum SentinelValue<A> {
  Sentinel,
  Value(A),
}
```

Does that make sense to you?

Reply:
yea it kind of sucks that we'd lose the nice compact representation of just an int though. definitely more typesafe though.

Reply:
Ya, maybe it could be something to look into for future improvements. I think using `Option` might be better as well, just for ease of existing functionality.

Reply:
The more I think of this the more I like this change :)